api/server.ts

```diff
@@ -2,38 +2,39 @@ import cors from 'cors';
 import dotenv from 'dotenv';
 import express from 'express';
 import { Server } from 'typescript-rest';
+
+const { ENV } = process.env;
+const path = `.env${ENV ? `.${ENV}` : ''}`;
+dotenv.config({ path });
+
+// NOTE: these imports happen here because they require dotenv to be configured first
+/* eslint-disable import/first */
+import { PORT, LOG_LEVEL, IS_TESTING } from './utils/constants';
 import createLogger, { initLogger } from './utils/logger';
-import { ENV, PORT, LOG_LEVEL } from './utils/constants';
+
+// NOTE: this initLogger function needs to be called before importing any file that uses the logger
```

Comment:
I don't think this is true - from my understanding as long as the logger an import is not _used_ before it is defined then it will have a value. This is just a javascript thing and is not specific to the logger. The tests would fail if there was an undefined import being used and they all were passing before you changed this back.

Reply:
PS: I was trying to get rid of the eslint disable stuff

Reply:
the problem's not that it would be undefined; `initLogger` sets the log level so anything that pulls in the logger before that's called will potentially have the wrong level set. (sorry that wasn't clear)

I'm sure there's a better way to accomplish this (could probably set the log level ad hoc?); you're welcome to investigate or we can make a ticket to make this simpler later

(PS sorry for committing straight to your branch, it's been a long weekend 💀)

Reply:
All I was saying is that it just doesn't need to change the import order because the log level will always be set at time of the _using_ the import, not at time of import.

Reply:
This is not true based on my manual testing (at least the way it's currently implemented)